### PR TITLE
Make counters atomic

### LINKED ISSUE #50: Make our counters atomic
... and get rid of the locks.

### PR DESCRIPTION
**Pull request by justinmueller05082000**
fixes #50

### CODE REVIEW

File: configobject/history/history.go [hunk 1]
```diff
@@ -34,23 +34,19 @@ var mysqlObservers = struct {
...
-	comment          int
-	flapping         int
-	acknowledgement  int
+	state            int64
+	notification     int64
+	usernotification int64
+	downtime         int64
+	comment          int64
+	flapping         int64
+	acknowledgement  int64
```
Comment (Al2Klimov) [inline configobject/history/history.go]: What datatype did we agree on for counters?

File: configobject/history/history.go [hunk 2]
```diff
@@ -34,23 +34,19 @@ var mysqlObservers = struct {
...
-var historyCounterLock = sync.Mutex{}
-
-func printAndResetHistoryCounter(counter *int, historyType string) {
+func printAndResetHistoryCounter(counter *int64, historyType string) {
 	if *counter > 0 {
 		log.Infof("Added %d %s history entries in the last 20 seconds", *counter, historyType)
-		historyCounterLock.Lock()
-		*counter = 0
-		historyCounterLock.Unlock()
+		atomic.SwapInt64(counter, 0)
```
Comment (Al2Klimov) [inline configobject/history/history.go]: Skywalker would say: Every line in this change is wrong.

File: configobject/history/history.go (hunk 1 above)
Comment (justinmueller05082000) [reply configobject/history/history.go]: uint64

File: configobject/history/history.go (hunk 2 above)
Comment (justinmueller05082000) [reply configobject/history/history.go]: should i undo it then

**Pull request merged**